ngrinder-controller/src/main/java/org/ngrinder/packages/PackageHandler.java

```diff
@@ -0,0 +1,188 @@
+
+	protected final int TIME_MILLIS_OF_DAY = 1000 * 60 * 60 * 24;
+	private final int EXEC = 0x81ed;
+
+	@Autowired
+	private Config config;
+
+	private Logger LOGGER = LoggerFactory.getLogger(PackageHandler.class);
+
+	protected Set<String> getDependentLibs(URLClassLoader urlClassLoader) {
+		Set<String> libs = new HashSet<>();
+		try (InputStream dependencyStream = urlClassLoader.getResourceAsStream(this.getDependenciesFileName())) {
+			final String dependencies = IOUtils.toString(dependencyStream);
+			for (String each : StringUtils.split(dependencies, ";")) {
+				libs.add(each.trim().replace("-SNAPSHOT", ""));
+			}
+		} catch (Exception e) {
+			LOGGER.error("Error while loading " + this.getDependenciesFileName(), e);
+		}
+		return libs;
+	}
+
+	protected void cleanUpPackageDir(boolean all) {
+		synchronized (AgentPackageService.class) {
+			final File packagesDir = getPackageDir();
+			final File[] files = packagesDir.listFiles();
+			if (files != null) {
+				for (File each : files) {
+					if (!each.isDirectory()) {
+						long expiryTimestamp = each.lastModified() + (TIME_MILLIS_OF_DAY * 2);
+						if (all || expiryTimestamp < System.currentTimeMillis()) {
+							FileUtils.deleteQuietly(each);
+						}
+					}
+				}
+			}
+		}
+	}
+
+	public void addBaseFolderToPackage(TarArchiveOutputStream tarArchiveOutputStream) throws IOException {
+		addFolderToTar(tarArchiveOutputStream, getBasePath());
+		addFolderToTar(tarArchiveOutputStream, getLibPath());
+	}
+
+	public void addFileToPackage(TarArchiveOutputStream tarArchiveOutputStream, File file, String path) throws IOException {
+		CompressionUtils.addFileToTar(tarArchiveOutputStream, file, getLibPath() + path);
+	}
+
+	public void addConfigToPackage(TarArchiveOutputStream tarOutputStream, Map<String, Object> agentConfigParam) throws IOException {
+		final String config = getConfigContent(agentConfigParam);
+		final byte[] bytes = config.getBytes();
+		addInputStreamToTar(tarOutputStream, new ByteArrayInputStream(bytes), getBasePath() + "__agent.conf",
+			bytes.length, TarArchiveEntry.DEFAULT_FILE_MODE);
+	}
+
+	public File getPackageFile(String regionName, String connectionIP, String ownerName, boolean forWindow) {
+		File packageDir = getPackageDir();
+		if (packageDir.mkdirs()) {
+			LOGGER.info("{} is created", packageDir.getPath());
+		}
+		final String packageName = getDistributionPackageName(regionName, connectionIP, ownerName, forWindow);
+		return new File(packageDir, packageName);
+	}
+
+	public void copyShellFile(TarArchiveOutputStream tarOutputStream) throws IOException {
+		ResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
+		Resource[] resources = resolver.getResources(getShellScriptsPath());
+		InputStream shellFileIs;
+		byte[] shellFileBytes;
+		for (Resource resource : resources) {
+			shellFileIs = resource.getInputStream();
+			shellFileBytes = IOUtils.toByteArray(shellFileIs);
+			IOUtils.closeQuietly(shellFileIs);
```

Comment:
I think shellFileIs should be wrapped by try catch so that resource leak never happen.